### PR TITLE
Make DDP and Horovod batch_size scaling examples explicit

### LINKED ISSUE #9784: Multi GPU training docs should show batch-size behaviour
I am fairly new to multi-GPU training, and it took me quite a while to understand that when using DDP the effective batch size scales with `num_gpus * batch_size`.
In fact, I thought that the batch-size that I provide to the lightning data module would be divided by the number of GPUs so that I often ran into memory errors.

Maybe this could be more clarified in the docs. https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html

### PR DESCRIPTION
**Pull request by tmachnitzki**
## What does this PR do?

Changes the docs to make DDP and Horovod batch size scaling behavior more explicit.

Fixes #9784 

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/advanced/multi_gpu.rst [hunk 1]
```diff
@@ -611,16 +611,17 @@ Let's say you have a batch size of 7 in your dataloader.
         def train_dataloader(self):
             return Dataset(..., batch_size=7)
 
-In (DDP, Horovod) your effective batch size will be 7 * gpus * num_nodes.
+In DDP and Horovod your effective batch size will be 7 * gpus * num_nodes.
```
Comment (rohitgr7) [inline docs/source/advanced/multi_gpu.rst:614]:
```suggestion
In DDP or Horovod your effective batch size will be 7 * gpus * num_nodes.
```

**Review by rohitgr7 — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
